core/src/main/java/org/apache/iceberg/BaseEntriesTable.java

```diff
@@ -283,19 +276,28 @@ static class ManifestReadTask extends BaseFileScanTask implements DataTask {
     private final ManifestFile manifest;
     private final Map<Integer, PartitionSpec> specsById;
 
+    ManifestReadTask(Table table, ManifestFile manifest, Schema projection, Expression filter) {
+      this(table.schema(), table.io(), table.specs(), manifest, projection, filter);
+    }
+
     ManifestReadTask(
-        Table table,
```

Comment:
did a bit refactoring. this constructor can be used by the JSON parser. the other constructor above is used for the previous purpose in the `planFiles` method above.